### PR TITLE
upgrade protobuf to ~2.0 and grpcio to 0.3.

### PR DESCRIPTION
**Pull request by hicqu**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,8 +7,11 @@ keywords = ["TiKV"]
...
+protobuf = "~2.0"
 futures = "0.1"
-grpcio = "0.2"
-raft = "0.2"
+grpcio = "0.3"
+
+# TODO: reset it to released version.
+[dependencies.raft]
+git = "https://github.com/pingcap/raft-rs"
+rev = "f271cd0dae4c868b2d92e19fe6c14e000598ea76"
```
Comment (siddontang) [inline Cargo.toml]: @BusyJay 

Can we release a new version for raft?

Comment (BusyJay) [reply Cargo.toml]: Yes. It will be released after the docs related pull requests are merged.

Comment (Hoverbear) [reply Cargo.toml]: https://github.com/pingcap/raft-rs/pull/69 if you feel like giving a LGTM. :)

Comment (Hoverbear) [reply Cargo.toml]: It is merged now. :)

**Comment by Hoverbear** [comment]
So all the changes outside the Cargo.toml are just generated code changes?

**Comment by BusyJay** [comment]
You need to update generate_rust.sh too.

**Comment by BusyJay** [comment]
raft 0.3.0 was released.

**Comment by siddontang** [comment]
please fix the conflict

**Comment by Hoverbear** [comment]
@hicqu Is this still DNM?

**Comment by BusyJay** [comment]
You need to check if the protobuf plugin's version match the minimal requirement.

**Pull request merged**